codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -66,412 +72,1106 @@
+                    // aggregate the flushes. This is done as ctx.flush() is expensive when as it may trigger an
+                    // write(...) or writev(...) operation on the socket.
+                    if (inFireChannelReadComplete) {
+                        flushPending = true;
+                    } else {
+                        flush0(ctx);
+                    }
+                } else {
+                    // There is nothing to flush so this is a NOOP.
+                }
+            }
+
+            @Override
+            public ChannelPromise voidPromise() {
+                return unsafeVoidPromise;
+            }
+
+            @Override
+            public ChannelOutboundBuffer outboundBuffer() {
+                // Always return null as we not use the ChannelOutboundBuffer and not even support it.
+                return null;
+            }
+        }
+
+        /**
+         * {@link ChannelConfig} so that the high and low writebuffer watermarks can reflect the outbound flow control
+         * window, without having to create a new {@link WriteBufferWaterMark} object whenever the flow control window
+         * changes.
+         */
+        private final class Http2StreamChannelConfig extends DefaultChannelConfig {
+
+            Http2StreamChannelConfig(Channel channel) {
+                super(channel);
+                setRecvByteBufAllocator(Http2StreamChannelRecvByteBufAllocator.INSTANCE);
+            }
+
+            /**
+             * @deprecated  Use {@link #getWriteBufferWaterMark()} instead.
+             */
+            @Override
+            @Deprecated
+            public int getWriteBufferHighWaterMark() {
+                return max(0, (int) min(Integer.MAX_VALUE, outboundFlowControlWindow));
+            }
+
+            /**
+             * @deprecated  Use {@link #getWriteBufferWaterMark()} instead.
+             */
+            @Override
+            @Deprecated
+            public int getWriteBufferLowWaterMark() {
+                return getWriteBufferHighWaterMark();
+            }
+
+            @Override
+            public MessageSizeEstimator getMessageSizeEstimator() {
+                return FlowControlledFrameSizeEstimator.INSTANCE;
+            }
+
+            @SuppressWarnings("deprecation")
+            @Override
+            public WriteBufferWaterMark getWriteBufferWaterMark() {
+                int mark = getWriteBufferHighWaterMark();
+                return new WriteBufferWaterMark(mark, mark);
+            }
+
+            @Override
+            public ChannelConfig setMessageSizeEstimator(MessageSizeEstimator estimator) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
+            @Deprecated
+            public ChannelConfig setWriteBufferHighWaterMark(int writeBufferHighWaterMark) {
+                throw new UnsupportedOperationException();
+            }
+
+            @Override
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Add the missing @deprecated Javadoc tag. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)